File: src/main/java/mnm/mods/util/gui/config/GuiSettingBoolean.java
package mnm.mods.util.gui.config;

import mnm.mods.util.config.Value;
import mnm.mods.util.gui.GuiCheckbox;
import mnm.mods.util.gui.config.GuiSetting.GuiSettingWrapped;

/**
 * A gui input for booleans as a checkbox.
 *
 * @author Matthew
 */
public class GuiSettingBoolean extends GuiSettingWrapped<Boolean, GuiCheckbox> {

    public GuiSettingBoolean(Value<Boolean> setting) {
        super(setting, new GuiCheckbox());
    }
}

File: src/main/kotlin/mnm/mods/tabbychat/client/gui/component/config/GuiSettingBoolean.kt
package mnm.mods.tabbychat.client.gui.component.config

import mnm.mods.tabbychat.client.gui.component.GuiCheckbox
import kotlin.reflect.KMutableProperty0

/**
 * A gui input for booleans as a checkbox.
 */
class GuiSettingBoolean(setting: KMutableProperty0<Boolean>) : GuiSetting.ValueSetting<Boolean, GuiCheckbox>(setting, GuiCheckbox())
